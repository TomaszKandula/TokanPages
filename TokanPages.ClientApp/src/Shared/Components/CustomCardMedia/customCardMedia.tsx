import * as React from "react";
import CardMedia from "@material-ui/core/CardMedia";
import validate from "validate.js";

export const renderCardMedia = (basePath: string, imageSource: string | undefined, className: string): JSX.Element =>
{
    return validate.isEmpty(imageSource) || validate.isEmpty(basePath)
        ? <div></div>
        : <CardMedia image={basePath + imageSource} className={className} />;
}
