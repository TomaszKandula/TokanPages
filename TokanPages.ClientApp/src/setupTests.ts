// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
