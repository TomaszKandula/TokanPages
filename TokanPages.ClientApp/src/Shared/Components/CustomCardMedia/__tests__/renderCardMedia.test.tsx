import "../../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { renderCardMedia } from "../customCardMedia";

describe("Test rendering 'Custom Card Media' component.", () => 
{
    interface IProperties 
    {
        basePath: string; 
        imageSource: string; 
        className: string;
    }

    const TestComponent = (props: IProperties): JSX.Element => 
    {
        return renderCardMedia(props.basePath, props.imageSource, props.className);
    }

    it("Should not render 'Custom Card Media' component.", () => 
    {
        const tree = shallow(<TestComponent 
            basePath=""
            imageSource=""
            className="style"
        />);
        expect(tree).toMatchSnapshot();
    });

    it("Should render 'Custom Card Media' component.", () => 
    {
        const tree = shallow(<TestComponent 
            basePath="http://domain.com/"
            imageSource="test-image.jpg"
            className="style"        
        />);
        expect(tree).toMatchSnapshot();
    });
});
