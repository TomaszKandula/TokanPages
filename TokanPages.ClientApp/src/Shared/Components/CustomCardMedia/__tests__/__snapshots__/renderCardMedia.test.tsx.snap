// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering 'Custom Card Media' component. Should not render 'Custom Card Media' component. 1`] = `<div />`;

exports[`Test rendering 'Custom Card Media' component. Should render 'Custom Card Media' component. 1`] = `
<WithStyles(ForwardRef(CardMedia))
  className="style"
  image="http://domain.com/test-image.jpg"
/>
`;
